engine/main.py

```diff
@@ -628,9 +629,12 @@ def unsubscribe_group(group_name, user):
     res = {'status':False}
     try:
         group = Group.objects.get(name=group_name)
-        membergroup = MemberGroup.objects.get(group=group, member=user)
-        membergroup.delete()
-        res['status'] = True
+        membergroups = MemberGroup.objects.filter(group=group)
+        membergroup = membergroups.get(member=user)
+
+        if confirm_unsubscribe(membergroups,user):
+            membergroup.delete()
+            res['status'] = True
```

Comment:
res['status'] = False is indicating there are errors occurred in a server. But the fact that user can't unsubscribe is not error but that's how we intend our system to be. I recommend using different key than _status_.